public/js/clients/firefox.js

```diff
@@ -102,7 +102,6 @@ function initPage(actions) {
 
   setupCommands({ threadClient, tabTarget, debuggerClient });
 
-  tabTarget.on("will-navigate", actions.willNavigate);
```

Comment:
I looked into `frameUpdate` and we definitely don't want to use that: https://github.com/mozilla/gecko-dev/blob/master/devtools/server/actors/webbrowser.js#L785

It's fired when any frame in the page changes. We should just use the navigation events -- why are we listening to `frameUpdate` at all? I can't remember, but I think it had something to do with refreshing while paused. Above it sounds like you are seeing both events fire when paused, so we should be able to just use the navigation events.